lib/beaker-benchmark/helpers.rb

```diff
@@ -42,61 +46,67 @@ def measure_perf_on(infrastructure_host, action_name, include_processes=false, p
           setup_atop(infrastructure_host)
           additional_args = ''
           additional_args = ',PRC,PRM,PRD' if include_processes
-          atop_cmd        = "sh -c 'nohup atop -P CPU,SWP,DSK#{additional_args} -i #{sample_interval} > #{atop_log} 2>&1 &'"
+          atop_cmd        = "sh -c 'nohup atop -P CPU,SWP,DSK#{additional_args} -i #{sample_interval} >  #{@atop_log} 2>&1 &'"
 
           on(infrastructure_host, atop_cmd)
           @beaker_benchmark_start = Time.now
         end
 
-        def stop_monitoring(infrastructure_host, action_name, atop_log_name, process_regex='.*')
-          if defined?@beaker_benchmark_start && !@beaker_benchmark_start.nil?
-            duration = Time.now - @beaker_benchmark_start
-          else
-            duration = nil
-          end
+        def stop_monitoring(infrastructure_host, process_regex='.*')
+          begin
+            if defined?@beaker_benchmark_start && !@beaker_benchmark_start.nil?
+              duration = Time.now - @beaker_benchmark_start
+            else
+              raise('No monitoring in progress, call start_monitoring before calling stop_monitoring')
+            end
 
-          # The atop process sticks around unless killed
-          on infrastructure_host, 'pkill -15 -f atop'
-          set_processes_to_monitor(infrastructure_host, process_regex)
-          parse_atop_log(infrastructure_host, action_name, duration, atop_log_name)
+            # The atop process sticks around unless killed
+            # It can also take some time to kill depending on how long it has been running and sampling rate.
+            retry_on infrastructure_host, 'pkill -15 -f atop', {:max_retries => 3, :retry_interval => 5}
+            set_processes_to_monitor(infrastructure_host, process_regex)
+            parse_atop_log(infrastructure_host, duration)
+          ensure
+            @beaker_benchmark_start = nil
+          end
         end
 
-        def parse_atop_log(infrastructure_host, action_name, duration, atop_log_name)
-          unless infrastructure_host.file_exist?(atop_log_name)
-            raise("atop log does not exist at #{atop_log_name}")
+        def parse_atop_log(infrastructure_host, duration)
+          unless infrastructure_host.file_exist?(@atop_log)
+            raise("atop log does not exist at #{@atop_log}")
           end
 
-          scp_from(infrastructure_host, atop_log_name, '/tmp')
+          log_dir = "#{TMP_DIR}/#{infrastructure_host.hostname}"
+          FileUtils::mkdir_p log_dir unless Dir.exist? log_dir
+          scp_from(infrastructure_host, @atop_log, log_dir)
           cpu_usage  = []
           mem_usage  = []
           disk_read  = []
           disk_write = []
 
-          process_cpu = []
           skip        = true
-          CSV.parse(File.read("/tmp/#{File.basename(atop_log_name)}"), { :col_sep => ' ' }) do |row|
+          CSV.parse(File.read(File.expand_path(File.basename(@atop_log), log_dir)), { :col_sep => ' ' }) do |row|
```

Comment:
don't need the file.basename any more since atop_log is a filename.  Might be nice to change the var name to something like atop_log_filename to make that clear.

Reply:
👍